CRM/Membershipextrasimporterapi/EntityImporter/Membership.php

```diff
@@ -0,0 +1,169 @@
+<?php
+
+class CRM_Membershipextrasimporterapi_EntityImporter_Membership {
+
+  private $rowData;
+
+  private $recurContributionId;
+
+  private $contactId;
+
+  private $cachedValues;
+
+  public function __construct($rowData, $contactId, $recurContributionId) {
+    $this->rowData = $rowData;
+    $this->contactId = $contactId;
+    $this->recurContributionId = $recurContributionId;
+  }
+
+  public function import() {
+    if (empty($this->rowData['membership_external_id'])) {
+      // todo : when the line item importer is created,
+      // we need to throw an exception if this field  is not set and line_item.entity_table = "civicrm_membership"
+      return NULL;
+    }
+
+    $membershipId = $this->getMembershipIdIfExist();
+    if ($membershipId) {
+      return $membershipId;
+    }
+
+    $sqlParams = $this->prepareSqlParams();
+    $sqlQuery = "INSERT INTO `civicrm_membership` (`contact_id` , `membership_type_id`, `join_date`, `start_date`, `end_date`, `status_id`,
+                 `is_pay_later`, `contribution_recur_id`, `is_override`, `status_override_end_date`) 
+            VALUES (%1, %2, %3, %4, %5, %6, %7, %8, %9, %10)";
+    CRM_Core_DAO::executeQuery($sqlQuery, $sqlParams);
+
+    $dao = CRM_Core_DAO::executeQuery('SELECT LAST_INSERT_ID() as membership_id');
+    $dao->fetch();
+    $membershipId = $dao->membership_id;
+
+    $sqlQuery = "INSERT INTO `civicrm_value_membership_ext_id` (`entity_id` , `external_id`) 
+           VALUES ({$membershipId}, %1)";
```

Comment:
Why use Complex (curly) syntax for $membershipId and mixed with executeQuery parameter?  Should it be `VALUES (%2, %1)";`? as  we can guarantee that  variables are safely escaped

Reply:
the membership Id is not a user input unlike the external Id, so I think it is fine.